cellfinder_core/main.py

```diff
@@ -37,7 +37,23 @@ def main(
     cube_height=50,
     cube_depth=20,
     network_depth="50",
+    *,
+    detect_callback=None,
+    classify_callback=None,
+    detect_finished_callback=None,
 ):
+    """
+    Parameters
+    ----------
+    detect_callback : Callable[int], optional
+        Called every time a plane has finished being processed during the
+        detection stage. Called with the plane number that has finished.
+    classify_callback : Callable[int], optional
+        Called every time tensorflow has finished classifying a point.
+        Called with the batch number that has just finished.
+    detect_finished_callback : Callable[list], optional
+        Called after detection is finished with the list of detected points.
+    """
```

Comment:
IMO it feels a bit odd to have a docstring which only describes certain parameters and not all

Reply:
this also occurs elsewhere

Reply:
It's a bit weird, but this can be added iteratively. If you guys document new additions, I can add in docs for original features in the future. Maybe add a `TODO` in areas with incomplete docs?

Reply:
Yeah, I think adding docstrings for other stuff is out of scope for this PR, but we should still document anything we're adding.

Reply:
You're both right, it's out of scope, but I think a TODO would be good